src/config/reader.rs

```diff
@@ -272,6 +274,17 @@ impl ConfigReader {
             path.join(src).to_string_lossy().to_string()
         }
     }
+
+    fn update_opentelemetry(&self, config_module: &mut ConfigModule) -> anyhow::Result<()> {
+        let server = &mut config_module.config.server;
+        let opentelemetry = &mut config_module.config.opentelemetry;
+
+        let runtime_ctx = TargetRuntimeContext { runtime: &self.runtime, vars: &server.vars };
```

Comment:
- Consider renaming, and relocating TargetRuntimeContext (nitpick)
- Don't store the complete runtime in this struct, just use vars and env.